lib/transformer.js

```diff
@@ -0,0 +1,143 @@
+'use strict'
+
+const zlib = require('zlib')
+const aws = require('aws-sdk')
+const {
+  checkFileFormat
+, hasProperty
+} = require('./utils.js')
+
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+module.exports = {
+  extractData
+, getLogs
+, prepareLogs
+, s3
+}
+
+function extractData(data) {
+  if (!data || !Object.keys(data).length) return []
+
+  const {
+    Records
+  , logFiles
+  , ...meta
+  } = data
+
+  const base = {
+    timestamp: Date.now()
+  }
+
+  let records = data
+  if (Records) {
+    records = Records
+    base.meta = meta
+  } else if (logFiles) {
+    records = logFiles
+    base.meta = meta
+  }
+
+  if (meta && meta.eventTime) {
+    base.timestamp = (new Date(meta.eventTime)).getTime()
+  }
+
+  if (!Array.isArray(records)) {
+    const lineJSON = {
+      line: records
+    , ...base
+    }
+
+    if (typeof records !== 'string') {
+      lineJSON.line = JSON.stringify(records)
+    }
+
+    return [lineJSON]
+  }
+
+  return records.map((record) => {
+    const lineJSON = {
+      line: record
+    , ...base
+    }
+
+    if (typeof record !== 'string') {
+      lineJSON.line = JSON.stringify(record)
+    }
+
+    if (record.eventTime) {
+      lineJSON.timestamp = (new Date(record.eventTime)).getTime()
+    }
+
+    return lineJSON
+  })
+}
+
+async function getLogs(params, callback) {
```

Comment:
no need for a callback here either, with an `async` function you'll want to throw errors or return the result

Reply:
Done!